Source/MVC6/Boilerplate.AspNetCore.TagHelpers/Twitter/TwitterCard.cs

```diff
@@ -5,35 +5,73 @@
     /// </summary>
     public abstract class TwitterCard : TagHelper
     {
         /// <summary>
-        /// The username attribute name.
+        /// The Twitter user ID of content creator.
         /// </summary>
-        protected const string UsernameAttributeName = "username";
+        protected const string CreatorTwitterIdAttributeName = "creator-id";
 
         /// <summary>
-        /// Gets the type of the Twitter card.
+        /// The Twitter @username of content creator e.g. @RehanSaeedUK.
         /// </summary>
-        /// <value>The type.</value>
-        public abstract TwitterCardType Type { get; }
+        protected const string CreatorUsernameAttributeName = "creator";
+
+        /// <summary>
+        /// The Twitter @username associated with the page e.g. @Microsoft. This is a required
+        /// property. Required for Twitter Card analytics.
+        /// </summary>
+        protected const string TwitterSiteAttributeUsername = "site";
+
+        /// <summary>
+        /// The Twitter Id associated with this site.
+        /// </summary>
+        protected const string TwitterSiteIdAttributName = "site-id";
+
+        /// <summary>
+        /// Gets or sets the Twitter user ID of content creator.
+        /// </summary>
+        /// <value>The Twitter user ID of content creator.</value>
+        [HtmlAttributeName(CreatorTwitterIdAttributeName)]
+        public string CreatorTwitterId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Twitter @username of content creator e.g. @RehanSaeedUK.
+        /// </summary>
+        [HtmlAttributeName(CreatorUsernameAttributeName)]
+        public string CreatorUsername { get; set; }
 
         /// <summary>
-        /// Gets or sets the Twitter username associated with the page e.g. @RehanSaeedUK. This is a required property.
+        /// Gets or sets the Site's Twitter site Id. Either twitter:site or twitter:site:id is required.
         /// </summary>
-        /// <value>The Twitter username associated with the page e.g. @RehanSaeedUK. This is a required property.</value>
-        [HtmlAttributeName(UsernameAttributeName)]
-        public string Username { get; set; }
+        /// <value>The twitter Id for the site.</value>
+        [HtmlAttributeName(TwitterSiteIdAttributName)]
+        public string TwitterSiteId { get; set; }
 
         /// <summary>
-        /// Synchronously executes the <see cref="TagHelper"/> with the given <paramref name="context"/> and
-        /// <paramref name="output"/>.
+        /// Gets or sets the twitter site username. e.g. @Microsoft.
+        /// </summary>
+        /// <value>
+        /// The Site's Twitter @username the card should be attributed to. Required for Twitter Card analytics.
+        /// </value>
+        [HtmlAttributeName(TwitterSiteAttributeUsername)]
+        public string TwitterSiteUsername { get; set; }
```

Comment:
Rename to `SiteUsername`. This will give us much shorter attribute names and the word Twitter will not be repeated everywhere.